src/main/kotlin/io/zerobase/smarttracing/resources/CreatorFilter.kt

```diff
@@ -0,0 +1,25 @@
+package io.zerobase.smarttracing.resources
+
+import io.zerobase.smarttracing.models.IdWrapper
+import javax.ws.rs.NameBinding
+import javax.ws.rs.container.ContainerRequestContext
+import javax.ws.rs.container.ContainerResponseContext
+import javax.ws.rs.container.ContainerResponseFilter
+import javax.ws.rs.core.HttpHeaders
+
+@NameBinding
+@Retention(AnnotationRetention.RUNTIME)
+@Target(AnnotationTarget.FUNCTION, AnnotationTarget.CLASS)
+annotation class Creator
+
+@Creator
+class CreatorFilter: ContainerResponseFilter {
+    override fun filter(req: ContainerRequestContext, resp: ContainerResponseContext) {
+        val entity = resp.entity
+        if (resp.status == 200 && entity is IdWrapper) {
+            resp.status = 201
+
+            resp.headers[HttpHeaders.LOCATION] = listOf(req.uriInfo.requestUriBuilder.path(entity.id.value).build())
```

Comment:
Haven't really worked with container response filter / container response text. I'm trying to understand this line - I'm unsure what it's exactly for. It looks like we have this @Creator annotation to check our posts that return type IdWrapper are returning OK responses and is an IdWrapper, but I'm not sure why updating headers is helpful.

Reply:
http/rest standard is that when a resource is created with a POST, the URL used to access the newly created resource should be specified in the `Location` header. jax-rs doesn't have a great way of doing this that abstracts out the HTTP. for example, if you just return an object from a handler method, it will automatically set the status code to 200. if you return null, it will automatically return 204. however, there isn't a way to cleanly map 201, so the normal way you do it is by having the method construct a `Response` that sets the status and location header.

i'm a big fan of the abstracting away of the http details in the business logic, so i wrote an annotation to take care of that http stuff without cluttering up the resource methods.